### DISCUSSION TITLE
Output folders depth standardization

### DISCUSSION THREAD

**Discussion by sbrunato** [CS-SI/eodag#446] [Polls]
For the moment, depending on the provider used to download products, you may have different output directory depth. This is due to additional top directories inside archives that are kept after extraction. 
Example of output directory:
```
.
├── .downloaded
└── S2A_MSIL1C_20220318T103751_N0400_R008_T31TCJ_20220318T142254
    └── S2A_MSIL1C_20220318T103751_N0400_R008_T31TCJ_20220318T142254.SAFE
        ├── AUX_DATA
        ├── DATASTRIP
        │   └── DS_VGS2_20220318T142254_S20220318T104246
        │       └── QI_DATA
        ├── GRANULE
        │   └── L1C_T31TCJ_A035183_20220318T104246
        │       ├── AUX_DATA
        │       ├── IMG_DATA
        │       └── QI_DATA
        ├── HTML
        └── rep_info
```
A parameter in the provider settings, `archive_depth` already exists to help eodag returning the good directory level that is effectively pointing to the data. But it may be something like `/path/to/{title}/{title}.SAFE` or  `/path/to/{title}.SAFE`. 
See issues where we have talked about it: https://github.com/CS-SI/eodag/issues/251 and https://github.com/CS-SI/eodag/issues/266 

In order to try standardizing output folders depth a bit more, would you prefer:
- to keep things as they are: having paths like `/path/to/{title}/{title}.SAFE` when `archive_depth=2` (like peps) or `/path/to/{title}` when `archive_depth=1` (like mundi)
- to delete an intermediate directory and have paths like `/path/to/{title}.SAFE` when `archive_depth=2` : we keep the archive root directory
- to delete an intermediate directory and have paths like `/path/to/{title}` when `archive_depth=2` : we delete the archive root directory

**Comment by floriandeboissieu** [comment]
Wouldn't it be a solution to have it defined in a parameter with the pattern? Example:
extraction_pattern = "{ouput_prefix}/{title}/{depth1}"
or
extraction_pattern = "{ouput_prefix}/{depth1}"
or
extraction_pattern = "{ouput_prefix}/{title}"
